src/createModuleProvider.js

```diff
@@ -75,6 +80,72 @@ module.exports = function () {
     return module.exports;
   };
 
+  /**
+   * Stored by reference path, holds the modules settings that need to be transformed
+   * into a dynamic URL.
+   * {
+   *  'referencePath': {
+   *     'setting.nested[4].path': 'relative-url'
+   *   }
+   * }
+   *
+   * @param settings - The settings to decorate
+   * @param filePaths - The list of paths down to each setting that needs to transform
+   * @param [moduleReferencePath] - If the filePaths come from a module, declare which one
+   */
+  var decorateSettingsWithDelegateFilePaths = function (
+    settings,
+    filePaths,
+    moduleReferencePath
+  ) {
+    // nothing to do
+    if (
+      !isDynamicEnforced ||
+      !settings ||
+      !Object.keys(settings).length ||
+      !Array.isArray(filePaths) ||
+      !filePaths.length
+    ) {
+      return settings;
+    }
+
+    var settingsCopy = objectAssign({}, settings);
```

Comment:
Not sure if this matters, but object assign will not do a deep clone copy.

Reply:
Yea. Should we introduce a `cloneDeep`? It would make [this feature](https://github.com/adobe/reactor-turbine/blob/master/src/rules/normalizeSyntheticEvent.js#L28-L30) more solid too.

Reply:
The only concern is about the speed. Do we need a deep clone?

Reply:
We're handing settings to extension developers, so they could modify references within the settings that would reflect across rules.

Reply:
We don't share settings objects between rules. Each delegate in each rule has its own settings object. Whoever receives a settings object inside his module is entitled to have access to it.